syne_tune/experiments/plotting.py

```diff
@@ -0,0 +1,664 @@
+            ("trial_id", 0),
+        ]
+        _check_and_set_defaults(new_params, default_values)
+        return ShowTrialParameters(**new_params)
+
+
+DEFAULT_XLABEL = "wall-clock time (secs)"
+
+
+@dataclass
+class PlotParameters:
+    """
+    Parameters specifying the figure.
+
+    In plots, lower is better. An original metric value ``metric_val`` is
+    converted as :code:`metric_multiplier * metric_val` if ``mode == "min"``,
+    and as :code:`1 - metric_multiplier * metric_val` if ``mode == "max"``.
+
+    :param metric: Name of metric, mandatory
+    :param mode: See above, "min" or "max". Defaults to "min" if not given
+    :param title: Title of plot. If ``subplots`` is used, see
+        :class:`SubplotParameters`
+    :param xlabel: Label for x axis. If ``subplots`` is used, this is
+        printed below each column. Defaults to :const:`DEFAULT_XLABEL`
+    :param ylabel: Label for y axis. If ``subplots`` is used, this is
+        printed left of each row
+    :param xlim: ``(x_min, x_max)`` for x axis. If ``subplots`` is used, see
+        :class:`SubplotParameters`
+    :param ylim: ``(y_min, y_max)`` for y axis.
+    :param metric_multiplier: See above. Defaults to 1
+    :param tick_params: Params for ``ax.tick_params``
+    :param aggregate_mode: How are values across seeds aggregated?
+
+        * "mean_and_ci": Mean and 0.95 normal confidence interval
+        * "median_percentiles": Mean and 25, 75 percentiles
+        * "iqm_bootstrap": Interquartile mean and 0.95 confidence interval
+          based on the bootstrap variance estimate
+
+        Defaults to :const:`DEFAULT_AGGREGATE_MODE`
+    :param dpi: Resolution of figure in DPI. Defaults to 200
+    :param grid: Figure with grid? Defaults to ``False``
+    :param subplots: If given, the figure consists of several subplots. See
+         :class:`SubplotParameters`
+    :param show_init_trials: See :class:`ShowTrialParameters`
+    """
+
+    metric: str = None
+    mode: str = None
+    title: str = None
+    xlabel: str = None
+    ylabel: str = None
+    xlim: Tuple[float, float] = None
+    ylim: Tuple[float, float] = None
+    metric_multiplier: float = None
+    tick_params: Dict[str, Any] = None
+    aggregate_mode: str = None
+    dpi: int = None
+    grid: bool = None
+    subplots: SubplotParameters = None
+    show_init_trials: ShowTrialParameters = None
+
+    def merge_defaults(self, default_params: "PlotParameters") -> "PlotParameters":
+        new_params = _impute_with_defaults(
+            original=self,
+            default=default_params,
+            names=[
+                "metric",
+                "mode",
+                "title",
+                "xlabel",
+                "ylabel",
+                "xlim",
+                "ylim",
+                "metric_multiplier",
+                "tick_params",
+                "aggregate_mode",
+                "dpi",
+                "grid",
+            ],
+        )
+        default_values = [
```

Comment:
This is because of this merging. You can specify *some* arguments when calling `plot`, the remaining ones should be taken from the default params passed at construction of `ComparativeResults`. If I use default values here, I cannot check whether you set something or not.